### PR TITLE
V2: Use output scaler for output scaling

### PR DESCRIPTION
**Pull request by hwpang**
## Description
Currently, the losses of training and validation sets are calculated using the scaled values. Then, the scaler is appended to the `loc` and `scale` attribute under `RegressionFFN` to calculate the losses of the test set. This is quite crumblesome and requires the user to remember only scale the training and validation set, but not the test set. Moreover, it is not intuitive for the user to compare the losses from training, validation, and test set.

## Example / Current workflow
Currently the test set is not scaled,
```
if isinstance(readout_ffn, RegressionFFN):
    scaler = train_dset.normalize_targets()     
    val_dset.normalize_targets(scaler)
    logger.info(f"Train data: loc = {scaler.mean_}, scale = {scaler.scale_}")
```
and it relies on register butter carefully at the right place (after evaluation is done on the validation step) in order to evaluate the test set as unscaled values.
```
if test_loader is not None:
        if args.task_type == "regression":
            model.predictor.register_buffer("loc", torch.tensor(scaler.mean_).view(-1, 1))
            model.predictor.register_buffer("scale", torch.tensor(scaler.scale_).view(-1, 1))
        results = trainer.test(model, test_loader)[0]
        logger.info(f"Test results: {results}")
```

## Bugfix / Desired workflow
This PR removes the `loc` and `scale` attributes from `RegressionFFN` and makes the unscale step explicitly by

```
if args.task_type == "regression":
                    preds = output_scaler.inverse_transform(preds)
```

This output scaler is saved to the model file.

## Relevant issues
https://github.com/chemprop/chemprop/issues/512

## Checklist
- [ ] linted with flake8?
- [ ] (if appropriate) unit tests added?

### CODE REVIEW

**Comment by shihchengli** [comment]
Thank you for submitting this PR. I think the code you wrote in lines 282-283 of cli/predict.py and lines 827-828 in cli/train.py addresses the issue with unscaled predictions.

Regarding the new class `InputScalers`, I think maintaining the original presentation as a dictionary is ok.

I agree with David on not integrating input/output scalers directly with the model. Upon reviewing the code, it appears that the rationale for incorporating input/output scalers into the model was to save those attributes in a checkpoint file. For those intending to use a checkpoint file to resume experiments, they should have access to the original training set to reconstruct the scalers externally. Conversely, when using the model file for inference, the scalers are already saved in the model file.
